packages/buffered/buffered.pony

```diff
@@ -0,0 +1,35 @@
+"""
+# Buffered Package
+
+The Buffered package provides two classes, `Writer` and `Reader`, for
+writing and reading messages using common encodings. These classes are
+useful when dealing with things like network data and binary file
+formats.
+
+## Example program
+
+```pony
+use "buffered"
+
+actor Main
+  new create(env: Env) =>
+    let reader = Reader
+    let writer = Writer
+
+    writer.u32_be(42)
+    writer.f32_be(3.14)
+
+    let b = recover iso Array[U8] end
+
+    for x in writer.done().values() do
+      b.append(x)
```

Comment:
I actually wanted to do `reader.append(x)`, but it doesn't quite work because the `append` takes an `Array[U8] val` but x is a `ByteSeq`. I had something like this earlier:

```pony
for x in writer.done.values() do
  match x
  | let s: String => reader.append(x.array())
  | let s: Array[U8] val => reader.append(x)
  end
end
```

But that felt not great.

If there's a better way to do this that I'm not seeing, please let me know.

I can make the change to `chunk`, I agree that it is much more descriptive.